### DISCUSSION TITLE
Minor Issues

### DISCUSSION THREAD

**Discussion by Eaclean** [Leantime/leantime#1361] [Q&A] [answered]
Hi,

I am having a couple of minor issues using Leantime installed on IIS like clicking "Do not show the tour screen again" isn't working and the "Forgot Password" page isn't emailing.

Looking at the logs I have one recurring message that I'm wondering might be the cause below.

PHP Parse error:  syntax error, unexpected 'appSettings' (T_STRING), expecting function (T_FUNCTION) or const (T_CONST) in E:\leantime\app\core\class.application.php on line 12

Anyone got any ideas about this error message?

Thanks

**Comment by iParisienne** [comment]
Hello, I am also running a leantime Installation on IIS and am experiencing the same problem.
Because of that I didn't want to open up a new discussion.

PHP: 8.2.1
IIS Version 10

To add onto that, the following minor problems still persist for me:

- After successful login, If I login and don't have any assigned projects, I get redirected to https://domain.com/leantime/leantime which results in a 404 error page (not browser 404, leantime 404 where I have the option to click on return to dashboard - which works)
- Notifications work, however, I cannot mark them as read - When I click on them, I don't get redirected and they don't get marked as read (see screenshot for the error I'm getting in console)
![Error_leantime](https://user-images.githubusercontent.com/96126061/216043403-0e5d8467-1ad3-42c5-bfec-ac05a87c3713.png)

The rest seems to work as far as I could test things out

**Reply by marcelfolaron** [reply]
Are you also seeing that appSettings error above?

**Reply by iParisienne** [reply]
No, I do not see that error; No PHP related errors at all. 
After several tests, I found that other features don't quite work either; for example, I can create milestones, but can't link a ToDo to them - Notes can't be saved in the milestone as well.
In addition, I deleted the user from whom I had the notification - but the notification is still there and is displayed as unread. Is this intentional behavior or does this have to do with the fact that I have not yet marked the notification as read? (Which currently does not work for me.)

According to the console it is always a 500 error.

I'm not back in the office until Monday, then I can edit appropriate screenshots to my post it if needed.

**Reply by iParisienne** [reply]
I didn't get any logs in \logs because of wrong write settings.
After modifying permissions, everything I've listed above creates the same error:

```
PHP Fatal error:  During class fetch: Uncaught ParseError: syntax error, unexpected '|', expecting variable (T_VARIABLE) in C:\inetpub\wwwroot\leantime\app\core\trait.eventhelpers.php:22
Stack trace:
#0 [internal function]: leantime\core\leantimeAutoloader()
#1 C:\inetpub\wwwroot\leantime\app\core\class.application.php(8): spl_autoload_call()
#2 C:\inetpub\wwwroot\leantime\app\core\class.autoload.php(48): require_once('C:\\inetpub\\wwwr...')
#3 [internal function]: leantime\core\leantimeAutoloader()
#4 C:\inetpub\wwwroot\leantime\public\index.php(24): spl_autoload_call()
#5 {main} in C:\inetpub\wwwroot\leantime\app\core\class.application.php on line 8
```

**EDIT:** I've managed to solve the above error - It was because of an incorrect configured Handler Mapping for the php-cgi.exe executable.
Everything seems to be working as intended so far.

The only problems which persist are:
- After login I get redirected to http://domain.com/leantime/leantime which results in a leantime specific 404 page (I Forgot to mention that this is a subfolder configuration)
- When clicking on notification for some items, like ToDos, the redirect works perfectly, however, for a split second I get the following screen:
![Error_leantime](https://user-images.githubusercontent.com/96126061/217521261-f43ef51a-52c1-4e65-bc9c-df11f738409e.png)
- Clicking "Do not show again" for the tour message doesn't work, as stated by OP.

**Reply by marcelfolaron** [reply]
For subfolder installations you need to follow the instructions in https://docs.leantime.io/#/installation/advanced

Basically you need to set appURL in your config file and update public/.htaccess to include line 4

**Reply by iParisienne** [reply]
appURL is already set.
Line 4 doesn't work with IIS, it does that automatically. All the other rewrite rules have been set. All thw redirects work as intended except that bizarre one after logging in.

**Reply by Eaclean** [reply]
@iParisienne Hi, have you tried the latest version? Thanks

**Reply by iParisienne** [reply]
I have updated to 2.3.11, yes. Seems to be working fine, altho I didn't test everything yet.

**Reply by Eaclean** [reply]
So you don't have any of the issues you mentioned above like clicking the "Do not show the tour screen again" etc?

Thanks

**Reply by iParisienne** [reply]
I don't seem to have that issue anymore after using and testing the new version.
For the wrong redirect after login I've hardcoded the URL in the login template, which works.

**Reply by Eaclean** [reply]
Great. Did you upgrade your existing instance or start from fresh?

**Reply by iParisienne** [reply]
I've upgraded the existing 2.3.6 instance to 2.3.11.

**Comment by Eaclean** [comment]
Are you using these rewrite rules?

<rule name="rule 1B">
	<match url="^/?$"  />
	<action type="Rewrite" url="/index.php?act=dashboard.show"  />
</rule>
<rule name="rule 2B">
	<match url="^([^/\.]+)/?$"  />
	<action type="Rewrite" url="/index.php?act={R:1}"  appendQueryString="true" />
</rule>
<rule name="rule 3B">
	<match url="^([^/\.]+)/([^/\.]+)/?$"  />
	<action type="Rewrite" url="/index.php?act={R:1}.{R:2}"  appendQueryString="true" />
</rule>
<rule name="rule 4B">
	<match url="^([^/\.]+)/([^/\.]+)/([^/\.]+)/?$"  />
	<action type="Rewrite" url="/index.php?act={R:1}.{R:2}&amp;id={R:3}"  appendQueryString="true" />
</rule>

**Reply by iParisienne** [reply]
I am using the rewrite rules from the public folder which I've converted into a web.config file.

**Reply by Eaclean** [reply]
Yea it's just those.

**Comment by marcelfolaron** [comment]
Can you show me the content of that file (E:\leantime\app\core\class.application.php )? It seems odd. The line defines a variable and this is the type definition. Did you make any modifications to the file?

**Reply by Eaclean** [reply]
The file has not been changed, to double check I redownloaded the the application and copied over the the file but it's the same message.

Thanks

**Comment by marcelfolaron** [comment]
Oh, what PHP version are you using? I just saw that this happens in PHP7.4
with functionality from PHP8.
We only support version 8 and up.

**Reply by Eaclean** [reply]
Apologies, I had a minor configuration issue regarding environment variables. This error is no longer showing.

However I still have the same issues as before like the other user.

I am using PHP 8.1.4.

Basically clicking on the "do not show tour" doesn't seam to work and no email being sent.

Getting no error messages at all now.

**Comment by marcelfolaron** [comment, marked as answer]
This is an error related to your PHP version.
It appears you are using php7? Leantime only supports php8.

**Reply by iParisienne** [reply]
Indeed it was, I've already made an edit to my comment with the remaining problems.
